### PR TITLE
[bugfix] fix unordered favorites

### LINKED ISSUE #1151: [bug] unordered Favorites
### Describe the bug with a clear and concise description of what the bug is.

favorite post list elements are not ordered on chronological descent order (as expected)

### What's your GoToSocial Version?

GoToSocial 0.5.2 git-c31f219

### GoToSocial Arch

_No response_

### Browser version

any

### What happened?

using pinafore, tusky, ... I went to review my _FAV list_ searching for a recent fav and noticed that it was way down the list.

asked about it in matrix room and I was informed that it was a old issue with low priority (that's fine for me) and invited to **open here an issue** so it can be tracked.

### What you expected to happen?

favorite post correctly time based ordered. Preferebly order by the moment user favorites them (in my opinion), not the original post time stamp.

### How to reproduce it?

go to your favorites and check

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by voigt**
# Description

This pull request fixes the ordering of favorites list.

closes #1151 (improves solution of #1236)

## Checklist

Please put an x inside each checkbox to indicate that you've read and followed it: `[ ]` -> `[x]`

If this is a documentation change, only the first checkbox must be filled (you can delete the others if you want).

- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.
- [x] I/we have commented the added code, particularly in hard-to-understand areas.
- [x] I/we have made any necessary changes to documentation.
- [X] I/we have added tests that cover new code.
- [x] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

File: internal/db/bundb/timeline.go [hunk 1]
```diff
@@ -233,7 +233,11 @@ func (t *timelineDB) GetFavedTimeline(ctx context.Context, accountID string, max
 
 	// Sort by favourite ID rather than status ID
 	slices.SortFunc(faves, func(a, b *gtsmodel.StatusFave) bool {
+<<<<<<< HEAD
 		return b.CreatedAt.Before(a.CreatedAt)
+=======
+		return a.ID > b.ID
+>>>>>>> 55357b89 ([bugfix] fix unordered favorites)
```
Comment (igalic) [inline internal/db/bundb/timeline.go:240]: yes looks like you pushed a merge conflict

**Review by igalic — CHANGES_REQUESTED** [review]

File: internal/db/bundb/timeline.go (hunk 1 above)
Comment (voigt) [reply internal/db/bundb/timeline.go]: my bad... fixed now.

**Comment by voigt** [comment]
I'm still unsure whether to provide a test. What exactly would you like to be seen tested in this case?

**Comment by tsmethurst** [comment]
> I'm still unsure whether to provide a test. What exactly would you like to be seen tested in this case?

Will respond to this later today, I've got some ideas :)

**Comment by voigt** [comment]
@tsmethurst Sure, let me know - happy to implement and improve testing! :)

**Comment by tsmethurst** [comment]
So I think a good way of more or less end-to-end testing this would be to just put some faves in the db, then retrieve them via the handler (so, a test in here somewhere: https://github.com/superseriousbusiness/gotosocial/tree/main/internal/api/client/favourites), and check that they're ordered correctly, and that the link headers are what we would expect. You can look around in some of the other api packages for some inspiration on testing, since we do something similar in a lot of places :) What do you think?

**Comment by voigt** [comment]
@tsmethurst I added a test, I hope this is what you expect. I added some test data, as you suggested, fetching the favorites of `local_account_1` and checking whether the first and last items are correct.

**Comment by tsmethurst** [comment]
Great stuff, thank you! Can you add the license text header to the new files (just copy and paste from another .go file)? Then this is ready to be merged imo :+1:

**Comment by voigt** [comment]
✅ Done!

**Pull request merged**